app/views/layouts/_header.html.erb

```diff
@@ -1,58 +1,95 @@
+    <%= image_tag("CircuitVerse.svg", class: "navbar-logo", alt: "CircuitVerse Logo") %>
   </a>
-  <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#collapsedNavbar"
-          aria-controls="collapsedNavbar" aria-expanded="false" aria-label="Toggle navigation">
+  <div class="row-text navbar-search-icon-container navbar-search-icon-oncollapse">
+    <i class="fa fa-search <%= request.path == '/search' ? 'active' : '' %>" tabindex="0" aria-label="search"></i>
+  </div>
+  <button class="navbar-toggler navbar-light" type="button" data-toggle="collapse" data-target="#collapsed-navbar"
+  aria-controls="collapsedNavbar" aria-expanded="false" aria-label="Toggle navigation">
     <span class="navbar-toggler-icon"></span>
   </button>
-
-  <div class="collapse navbar-collapse" id="collapsedNavbar">
-    <ul class="navbar-nav ml-auto mt-2 mt-lg-0 navigationBar-item">
-      <li class="nav-item px-1">
-        <a class="btn btn-outline-success rounded-0 simulator text-dark " href="/simulator">Simulator</a>
+  <div class="collapse navbar-collapse" id="collapsed-navbar">
+    <ul class="navbar-nav ml-auto navigationBar-item">
+      <li class="nav-item px-1 pt-1 navbar-search-icon-onexpand">
+        <div class="row-text navbar-search-icon-container">
+          <i class="fa fa-search <%= request.path == '/search' ? 'active' : '' %>" tabindex="0" aria-label="search"></i>
+        </div>
       </li>
       <li class="nav-item px-1">
-        <a class="nav-link text-dark" href="https://learn.circuitverse.org/">Learn</a>
+        <a class="btn rounded-0 navbar-simulator-text" href="/simulator">Simulator</a>
       </li>
+      <div class="navbar-nav nav-item dropdown">
+        <li class="nav-item px-1 dropdown-toggle-hidden">
+          <a class="nav-link dropdown-toggle navbar-item navbar-text" id="navbar-dropdown-1" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" tabindex="0" aria-label="getting started">Getting Started</a>
+          <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbar-dropdown-1">
+            <a class="dropdown-item" href="https://learn.circuitverse.org/">Learn</a>
+            <a class="dropdown-item" href="https://docs.circuitverse.org/">Documentation</a>
+          </div>
+        </li>
+      </div>
       <li class="nav-item px-1">
-        <a class="nav-link text-dark" href="https://docs.circuitverse.org/">Documentation</a>
-      </li>
-      <li class="nav-item px-1 <%= request.path == '/examples' ? 'active' : '' %>">
-        <a class="nav-link text-dark" href="/examples">Examples</a>
-      </li>
-      <li class="nav-item px-1 <%= request.path == '/contribute' ? 'active' : '' %>">
-        <a class="nav-link text-dark" href="/contribute">Contribute</a>
+        <a class="nav-link navbar-item navbar-text <%= request.path == '/features' ? 'active' : '' %>" href="/features">Features</a>
```

Comment:
Target link for Features nav link should be the Features section on the homepage.